src/index.ts

```diff
@@ -1,17 +1,75 @@
+import * as fs from 'fs';
+import * as https from 'https';
+import * as path from 'path';
+
 const core = require('@actions/core');
 const exec = require('@actions/exec');
 
 import buildExec from './buildExec';
 
-const {execArgs, options, failCi} = buildExec();
-
-exec.exec('bash', execArgs, options)
-    .catch((err) => {
-      if (failCi) {
-        core.setFailed(
-            `Codecov failed with the following error: ${err.message}`,
-        );
-      } else {
-        core.warning(`Codecov warning: ${err.message}`);
-      }
-    });
+let failCi;
+const setFailure = (message, failCi) => {
+   failCi ? core.setFailed(message) : core.warning(message);
+};
+const isWindows = (platform) => {
+  return platform === 'windows';
+};
+const PLATFORMS = ['alpine', 'linux', 'macos', 'windows'];
+
+try {
+  const {execArgs, options, failCi, platform} = buildExec();
+  if (!PLATFORMS.includes(platform)) {
+    setFailure(
+        `Codecov: Encountered an unexpected platform: ${platform}`,
+        failCi,
+    );
+    process.exit();
+  }
+  const url = `https://uploader.codecov.io/latest/codecov-${platform}${isWindows(platform) ? '.exe' : ''}`;
```

Comment:
This can be moved into a separate function in order to abstract its domain complexities:

```suggestion
  const filename = getFilenameFromPlatform(platform);
  const url = `https://uploader.codecov.io/latest/${filename}`;
```

Reply:
```ts
function getFilenameFromPlatform(platform) {
  if (!PLATFORMS.includes(platform)) {
    throw new Error(`Codecov: Encountered an unexpected platform: ${platform}`);
  }
  
  if (platform === "windows") {
    return "codecov.exe";
  } else {
    return `codecov-${platform}`;
  }
}
```